app/handlers/newflow/student_signup.rb

```diff
@@ -14,11 +23,83 @@ def authorized?
     end
 
     def required_params
-      @required_params ||= [:email, :first_name, :last_name, :password, :role]
+      @required_params ||= [:email, :first_name, :last_name, :password]
     end
 
     def handle
-      super
+      new_login_credential_check
+      return if errors?
+
+      outputs.email = signup_params.email
```

Comment:
is all of this unit tested somewhere?

Reply:
Same as https://github.com/openstax/accounts/pull/837#discussion_r428295270